docs/assets/sphinx_rsts/modules/core/lattice/hal.rst

```diff
@@ -0,0 +1,39 @@
+.. _hal:
+
+Core Lattice Hardware Abstraction Layer Documentation
+======================================================
+
+.. mermaid::
+
+  graph BT
+      a[DCRTPolyInterface] --> |Inherited by|b[DCRTPoly - HAL Default];
+      b[DCRTPoly-HAL Default] --> |Inherited by|c[HexlDCRTPoly];
+
+This hardware abstraction layer allows OpenFHE to use a variety of device backends while still allowing for high performance. As of March 31st 2022, we have the default backend, and the `Intel Hexl <https://github.com/intel/hexl>`_ backend.
```

Comment:
While I'm not sure we want to specify hardware specifically, I feel hardware-integrated backends may be clearer towards that point here.